src/briefcase/commands/dev.py

```diff
@@ -234,22 +259,27 @@ def __call__(
         if not run_app:
             # If we are not running the app, it means we should update requirements.
             update_requirements = True
-        if update_requirements or not dist_info_path.exists():
-            self.console.info("Installing requirements...", prefix=app.app_name)
-            self.install_dev_requirements(app, **options)
-            write_dist_info(app, dist_info_path)
-
-        if run_app:
-            if app.test_mode:
-                self.console.info(
-                    "Running test suite in dev environment...", prefix=app.app_name
+
+        with self.get_venv_context(
```

Comment:
Looks like this hasn't been updated - which is then being masked by the mocking in the tests.